java/vector/src/main/java/org/apache/arrow/vector/BaseVariableWidthVector.java

```diff
@@ -1373,6 +1373,9 @@ public int hashCode(int index) {
 
   @Override
   public int hashCode(int index, ArrowBufHasher hasher) {
+    if (isNull(index)) {
+      return 0;
```

Comment:
is zero really a good value to return for hashing purposes?

Reply:
According to my experience, zero is commonly used as hash code for nulls.

Reply:
I also find this in "Effective Java, 3rd ed." pp. 51:

If the value of the field is null, use 0 (or some other constant, but 0 is traditional).